src/uucp_break.ml

```diff
@@ -35,6 +35,38 @@ let sentence u = Array.unsafe_get Low.sentence_of_int (Low.sentence u)
 
 let east_asian_width u = Uucp_rmap.get Uucp_break_data.east_asian_width_map u
 
+let err_ctrl_char e =
+  invalid_arg (Printf.sprintf "0x%02X is a control character" e)
+
+let err_not_uchar e =
+  invalid_arg (Printf.sprintf "U+%04X is not a unicode scalar value." e)
+
+let terminal_width_hint =
+  let gc = Uucp_gc.general_category in
+  function
+  (* C0 (without 0x00) + DELETE + C1 is an error. *)
+  |u when 0 < u && u <= 0x1f || 0x7f <= u && u <= 0x9f -> err_ctrl_char u
```

Comment:
> It's just that an exception + precondition is cheaper to the client sanitizing its input,

What I fear a bit is that clients will tend to simply fold over the scalar values of arbitrary strings they read e.g. from a file and that we will see more `Invalid_argument` exception being thrown in our face than needed. 

I prefer to have the function total over scalar values as it's the way all the other function work in the library. Using an option will remind clients that they could end up having data for which the computation won't make sense. We should still precisely document in which cases `None` is returned though (as is done by the PR).